e2e/pages/vm-sidebar.po.ts

```diff
@@ -171,4 +172,142 @@ export class VMSidebar extends CloudstackUiPage {
       by.cssContainingText('.value', expected),
     );
   }
+
+  clickChangeSO() {
+    const edit = element(by.css('cs-service-offering-details .mat-icon-button'));
+    const EC = protractor.ExpectedConditions;
+    browser.wait(EC.elementToBeClickable(edit), 2000, 'No edit Group button').then(() => {
+      edit.click();
+      this.waitDialogModal();
+    });
+  }
+
+  selectFixedTab() {
+    element
+      .all(by.css('cs-service-offering-dialog-container mat-button-toggle-group mat-button-toggle'))
+      .first()
+      .click();
+  }
+
+  changeSO() {
+    element(
+      by.xpath(
+        "//mat-radio-button[@class='mat-radio-button mat-accent mat-radio-checked']/ancestor::tbody//mat-radio-button[@class='mat-radio-button mat-accent']",
+      ),
+    ).click();
+  }
+
+  getSelectedSO() {
+    return element(
+      by.xpath(
+        "//mat-radio-button[@class='mat-radio-button mat-accent mat-radio-checked']/ancestor::td/preceding-sibling::td//span",
+      ),
+    ).getText();
+  }
+
+  getChangedSO() {
+    return element(
+      by.xpath(`//cs-service-offering-details//div[text()="Name"]/following-sibling::div`),
+    ).getText();
+  }
+
+  checkWarning() {
+    element(by.css('.message.warning.ng-star-inserted')).isPresent();
+    return element(by.css('.message.warning.ng-star-inserted')).getText();
+  }
+
+  isEnabledChangeSO() {
+    expect(
+      element(by.css('cs-service-offering-details button[ng-reflect-disabled=true]')),
+    ).toBeTruthy('Change SO button is enabled');
+  }
+
+  isEnabledAddAffinityGroup() {
+    expect(element(by.css('cs-affinity-group button[ng-reflect-disabled=true]'))).toBeTruthy(
+      'Affinity group button is enabled',
+    );
+  }
+
+  isEnabledAddSSHKey() {
+    expect(element(by.css('cs-vm-ssh-keypair button[ng-reflect-disabled=true]'))).toBeTruthy(
+      'Add SHH Key is enabled',
+    );
+  }
+
+  clickChangeSSHKey() {
+    const edit = element(by.css('cs-vm-ssh-keypair button'));
+    const EC = protractor.ExpectedConditions;
+    browser.wait(EC.elementToBeClickable(edit), 2000, 'No edit Group button').then(() => {
+      edit.click();
+      this.waitDialogModal();
+    });
+  }
+
+  selectSSHKey(index) {
+    element(by.css('cs-ssh-keypair-reset mat-select')).click();
+    element
+      .all(by.css('mat-option'))
+      .get(index)
+      .click();
+    return element
+      .all(by.css('mat-option span'))
+      .get(index)
+      .getText();
+  }
+
+  getSHHKey() {
+    return element(by.css('cs-vm-ssh-keypair span')).getText();
+  }
+
+  checkResetWarning() {
+    return browser.wait(
+      protractor.ExpectedConditions.presenceOf(element(by.css('.mat-dialog-content.notification'))),
+      5000,
+    );
+  }
+
+  isEnabledCreateAffGroupButton() {
+    return element(by.css('.add-rule-button.mat-icon-button')).isEnabled();
+  }
+
+  setIncorrectAffGroupName(name) {
```

Comment:
This function sets any entered name for an affinity group, not just an incorrect one.

Change the name of the function.